### ISSUE TITLE
KITTI DetectNet example: not detecting pedestrians

### ISSUE THREAD

**Issue by DarylWM** [NVIDIA/DIGITS#1319]
I've been trying to use the KITTI dataset and the DetectNet example to detect pedestrians, but after 100 epochs (batch size 8) the mAP remains zero. I've set custom classes to:
dontcare,pedestrian

I haven't changed anything else from the example. For example, I haven't set the transformations in the augmentation parameters to zero. Should I need to?

**Comment by JorgeVilchis** [comment]
Did make sure to set subtract mean to none when creating the training job in DIGITS ?

**Comment by DarylWM** [comment]
Yes, "Subtract Mean" is set to None.

Zero mAP even after 150 epochs.

**Comment by DarylWM** [comment]
It's interesting that mAP increases above zero at epoch 3 when it's learning to detect cyclists, where custom classes is:
dontcare,Cyclist

**Comment by gheinrich** [comment]
If you don't have https://github.com/NVIDIA/DIGITS/commit/8a9b50b055ffd1ef24c898fc74c01e7026003ba3#diff-1771beb4c58b8620538518e01e546321 you need to lower case your custom class names.

**Comment by DarylWM** [comment]
Thanks @gheinrich, I have that change. My problem is I can't get the model to detect pedestrians; cars and cyclists are detected, but not pedestrians.

**Comment by d-ohm** [comment]
Daryl - Did you set the class mapping in your network description to include pedestrians? See the definitions in the link below. Make sure you edit your custom network before training. I believe the sections you want to edit will look something like this

detectnet_groundtruth_param {
    stride: 16
    scale_cvg: 0.40000000596
    gridbox_type: GRIDBOX_MIN
    min_cvg_len: 20
    coverage_type: RECTANGULAR
    image_size_x: 1248
    image_size_y: 384
    obj_norm: true
    crop_bboxes: true
    object_class {
      src: 1
      dst: 0
    }

https://github.com/NVIDIA/DIGITS/tree/digits-5.0/digits/extensions/data/objectDetection#custom-class-mappings

**Comment by eweill** [comment]
I am also trying to accomplish the same task and have tried many different configurations of data set creation (for object detection) as well as modeling styles (different learning rate strategies, etc.).  I am having the same problem where I can't get the model to detect pedestrians at all but cars get detected perfectly.  No matter if I use a single class pedestrian detector or a 2-class car-pedestrian detector, the pedestrians are never detected.

**Comment by DarylWM** [comment]
I tried turning off the augmentation parameters just to see if that might help, but it didn't. Any ideas what the difference is between cyclists (which can be detected) and pedestrians (which can't)?

Pedestrians: 4299 in the training set, 188 in the validation set
Cyclists: 1445 in the training set, 182 in the validation set

Could it be that the variation of pedestrians is greater than for cyclists?

**Comment by s5919** [comment]
Have you got any new results? 

I am working at the same problem: Pedestrian Detection with DetectNet. At first I thought the KITTI database hasn't got enough or bad pedestrian examples. Therefore I created my own dataset, which is labeled like the KITTI dataset. Unfortunately mAP stays at zero during the training. 

Is it possible, that the network is not able to detect pedestrians?

**Comment by alpop** [comment]
I've got same problem, too. No clue how to debug this....

**Comment by DarylWM** [comment]
Thanks for the tip @alpop, I'll give that a try.

By the way, for the sake of comparison, I tried Darknet on the pedestrians in the Kitti data set.

![image](https://cloud.githubusercontent.com/assets/1016303/21753912/71b49c3e-d64a-11e6-99d6-ca9aa8ebd181.png)

**Comment by JorgeVilchis** [comment]
Nice job ! Do you think the performance in terms of localization / classification is better using DarkNet ?

**Comment by ShervinAr** [comment]
@DarylWM  did you manage to get pedestrians detected?

**Comment by DarylWM** [comment]
Not yet with DetectNet.

**Comment by alpop** [comment]
Great job, nice picture! What it should look like with DectectNet (on held-out image with map around 20)
![pedestrians](https://cloud.githubusercontent.com/assets/11949491/22048187/5717eacc-dd3c-11e6-8a90-5150e009feef.png)

**Comment by ShervinAr** [comment]
@alpop  could you please post also the original image without bounding boxes so that I could test it with my trained net?

**Comment by sulth** [comment]
Pedestrian for single class and two class not getting learned.Please someone give me some tips.Also is the darknet approach good and accurate?

**Comment by sulth** [comment]
@DarylWM  have you got detectnet multiclass detection run.May i also know if darknet better than detectnet
